File: 1768. Merge Strings Alternately/Solution.py
word1 = "ab"
word2 = "pqrs"

class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        mergedString = ""
        word1_len = 0
        word2_len = 0
        while word1_len < len(word1) or word2_len < len(word2):
            if word1_len < len(word1):
                mergedString += word1[word1_len]
                word1_len += 1
            if word2_len < len(word2):
                mergedString += word2[word2_len]
                word2_len += 1
        return mergedString

        
solution = Solution()
print(solution.mergeAlternately(word1, word2))

File: 1768. Merge Strings Alternately/Solution.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        StringBuilder out = new StringBuilder();
        int p1 = 0;
        int p2 = 0;

        while (p1 < word1.length() || p2 < word2.length()) {
            if (p1 < word1.length()) {
                out.append(word1.charAt(p1));
                p1++;
            }
            if (p2 < word2.length()) {
                out.append(word2.charAt(p2));
                p2++;
            }
        }

        return out.toString();
    }

    public static void main(String[] args) {
        String word1 = "ab";
        String word2 = "pqrs";
        Solution solution = new Solution();
        String merged = solution.mergeAlternately(word1, word2);
        System.out.println(merged);
    }
}